### PR TITLE
Widget tests

### PR DESCRIPTION
**Pull request by tomerlichtash**
## Related Issues

Fixes <link_to_github_issue>

## Related PRs

| branch       | PR         |
| ------------ | ---------- |
| service a PR | Link to PR |
| service b PR | Link to PR |

## Description

A few sentences describing the overall goals of the pull request's commits.

## Must

- [ ] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

File: packages/user-management-widget/test/user-management-widget.test.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
+import '@testing-library/jest-dom';
+import { waitFor } from '@testing-library/dom';
+import { apiPaths } from '../src/lib/widget/api/apiPaths';
+import { mockUsers } from './mocks/mockUsers';
+import { createSdk } from '../src/lib/widget/api/sdk';
+import { pluralize } from '../src/lib/helpers/generic';
+import '../src/lib/index';
+// import '@descope/web-components-ui';
+
+// globalThis.DescopeUI = {
```
Comment (tomerlichtash) [inline packages/user-management-widget/test/user-management-widget.test.ts:10]: If we're not going to attend these tests in jest we can remove these comments

File: packages/user-management-widget/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    // },
+  ],
+
+  /* Run your local dev server before starting the tests */
+  webServer: [
+    {
+      command: 'npx serve node_modules/@descope/web-components-ui/dist -p 8765',
+    },
+    {
+      command: 'npx serve build -l 5555',
```
Comment (nirgur) [inline packages/user-management-widget/playwright.config.ts:83]: Let's make sure it's running on a clean env

File: packages/user-management-widget/e2e/user-management-widget.spec.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    // submit email
+    await createUserEmailInput.fill('someEmail@test.com');
+
+    // click modal create button
+    await createUserButton.click();
+
+    // update grid items
+    await expect(
+      page.locator(`text=${mockNewUser['loginIds'][0]}`).first(),
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts:99]: Added case in mock

File: packages/user-management-widget/e2e/user-management-widget.spec.ts [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    expect(deleteUserModal).toBeVisible();
+
+    // click modal delete button
+    await deleteUserModalButton.click();
+
+    // wait for modal to close
+    await page.waitForTimeout(500);
+
+    // delete modal closed
+    await expect(page.locator('Delete Users')).toBeHidden();
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts:135]: WDYM reuse? Each test locates its own locators; changed selectors though.

File: packages/user-management-widget/e2e/user-management-widget.spec.ts [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    await expect(
+      page.locator('text=2 users deleted successfully'),
+    ).toBeVisible();
+
+    // update grid items
+    await expect(page.locator('#items')).toBeEmpty();
+  });
+
+  test('search users', async ({ page }) => {
+    const searchInput = page
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts:147]: added test-id instead

File: packages/user-management-widget/e2e/user-management-widget.spec.ts [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+    // enter search string
+    await searchInput.fill('user2');
+
+    // wait for results to filter
+    await page.waitForTimeout(500);
+
+    // only search results shown in grid
+    await expect(page.locator('text=user1@user1.com').nth(0)).toBeHidden();
+    await expect(page.locator('text=user1@user1.com').nth(1)).toBeHidden();
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts:163]: One as e-mail, the other as loginId.

File: packages/user-management-widget/playwright.config.ts (hunk 2 above)
Comment (tomerlichtash) [reply packages/user-management-widget/playwright.config.ts:83]: Now it does. After deleting `build` it generates its again and serves it.

File: packages/user-management-widget/e2e/user-management-widget.spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (nirgur) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts:147]: Not sure if this is a good solution when we will have an editor

File: packages/user-management-widget/e2e/user-management-widget.spec.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (nirgur) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts:163]: This is error prone
IMO, it's better to use different values

File: packages/user-management-widget/e2e/user-management-widget.spec.ts [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+    // delete modal closed
+    await expect(page.locator('Delete Users')).toBeHidden();
+
+    // show notification
+    await expect(
+      page.locator('text=2 users deleted successfully'),
+    ).toBeVisible();
+
+    // update grid items
+    await expect(page.locator('#items')).toBeEmpty();
```
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts]: Agreed. In this case couldn't find an alternative.

File: packages/user-management-widget/e2e/user-management-widget.spec.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (tomerlichtash) [reply packages/user-management-widget/e2e/user-management-widget.spec.ts]: Fixed.

**Comment by tomerlichtash** [comment]
@nirgur All locators were changed to their optimal non-implementation-detail form. This means most of them use test-ids when possible, as we discussed. Why would this be a problem for a future editor?

Also, _some_ implementation details are still using locators, for no apparent alternative.

LMK what you think.

**Pull request merged**